### ISSUE TITLE
HMAC support in the dynamic format / compiler

### ISSUE THREAD

**Issue by kholia** [magnumripper/JohnTheRipper#1903]
@jfoug it would be super awesome to support various HMAC constructions in the dynamic format / compiler.

**Comment by jfoug** [comment]
Please help define the language syntax.

**Comment by jfoug** [comment]
Do we want HMAC, or just PBKDF2-HMAC-*  ?

**Comment by kholia** [comment]
@jfoug it will be nicer to get both, but HMAC looks like the common starting point.

I am not very familiar with the existing language, yet!

Something like `hmac-sha256(hmac-sha1($p, $s), $s)`?

**Comment by kholia** [comment]
More importantly, it would be nice to support long salts. Our existing HMAC-SHA256 format is currently limited to length 55 which is quite limiting in practice.

**Comment by magnumripper** [comment]
Yes, that's #1904 and I intend to have a look at it (eventually). I did it with some OpenCL format and it was easier than I feared - and no extra code in inner loop.
